packages/map-template/src/components/MapTemplate/MapTemplate.jsx

```diff
@@ -88,6 +86,133 @@ function MapTemplate({ apiKey, gmApiKey, mapboxAccessToken, venue, locationId, p
+            miSdkApiTag.setAttribute('type', 'text/javascript');
+            miSdkApiTag.setAttribute('src', 'https://app.mapsindoors.com/mapsindoors/js/sdk/4.23.1/mapsindoors-4.23.1.js.gz');
+            document.body.appendChild(miSdkApiTag);
+            miSdkApiTag.onload = () => {
+                setMapsindoorsSDKAvailable(true);
+                resolve();
+            }
+        });
+    }
+
+    /**
+     * Wait for the MapsIndoors JS SDK to be initialized, then react on changes on the props.
+     * React on changes in the MapsIndoors API key by fetching the required data.
+     */
+    useEffect(() => {
+        initializeMapsIndoorsSDK().then(() => {
+            const mapsindoors = window.mapsindoors;
+
+            setApiKey(apiKey);
+            setMapReady(false);
+            mapsindoors.MapsIndoors.setMapsIndoorsApiKey(apiKey);
+
+            Promise.all([
+                // Fetch all Venues in the Solution
+                mapsindoors.services.VenuesService.getVenues(),
+                // Fixme: Venue Images are currently stored in the AppConfig object. So we will need to fetch the AppConfig as well.
+                mapsindoors.services.AppConfigService.getConfig(),
+                // Ensure a minimum waiting time of 3 seconds
+                new Promise(resolve => setTimeout(resolve, 3000))
+            ]).then(([venuesResult, appConfigResult]) => {
+                venuesResult = venuesResult.map(venue => {
+                    venue.image = appConfigResult.venueImages[venue.name.toLowerCase()];
+                    return venue;
+                });
+                setAppConfigResult(appConfigResult);
+                setVenues(venuesResult);
+            });
+            setMapReady(false);
+        });
+    }, [apiKey]);
+
+    /*
+     * React to changes in the gmApiKey and mapboxAccessToken props.
+     */
+    useEffect(() => {
+        if (mapsindoorsSDKAvailable) {
+            setMapboxAccessToken(mapboxAccessToken);
+            setGmApyKey(gmApiKey);
+        }
+    }, [gmApiKey, mapboxAccessToken, mapsindoorsSDKAvailable]);
+
+    /*
+     * React on changes in the app user roles prop.
+     */
+    useEffect(() => {
+        if (mapsindoorsSDKAvailable) {
+            window.mapsindoors.services.SolutionsService.getUserRoles().then(userRoles => {
+                const roles = userRoles.filter(role => appUserRoles?.includes(role.name));
+                window.mapsindoors.MapsIndoors.setUserRoles(roles);
+            });
+        }
+    }, [appUserRoles, mapsindoorsSDKAvailable]);
+
+    /*
+     * React on changes in the externalIDs prop.
+     * Get the locations by external IDs, if present.
+     */
+    useEffect(() => {
+        if (externalIDs && mapsindoorsSDKAvailable) {
+            window.mapsindoors.services.LocationsService.getLocationsByExternalId(externalIDs).then(locations => {
+                setFilteredLocationsByExternalID(locations);
+            });
+        } else {
```

Comment:
Do we really want to call the set function when there is no mapsindoors SDK available?
I think everything could be wrapped in a check for `mapsindoorsSDKAvailable`, just like the useEffect for locationId.

Reply:
Good idea. I refactored the useEffect for the external IDs.